codegen/smithy-aws-ruby-codegen/src/main/java/software/amazon/smithy/aws/ruby/codegen/AWSConfig.java

```diff
@@ -43,5 +43,32 @@ public final class AWSConfig {
                     .build())
             .build();
 
+    public static final ClientConfig DUALSTACK = ClientConfig.builder()
+            .name("use_dualstack_endpoint")
```

Comment:
I think we should change these options to either just dualstack/fips or drop use_ prefix

Reply:
I debated it - I wanted to reduce friction in v3 -> v4 migration by keeping configuration options the same.  And additionally the ENV and shared config names match "use_X_endpoint" so wanted to keep that consistent.